test/FirebaseArduino_test.cpp

```diff
@@ -17,26 +17,82 @@
 #include "FirebaseObject.h"
 #include "gtest/gtest.h"
 
-TEST(FirebaseObjectTest, JsonLiteral) {
-  EXPECT_EQ(bool(FirebaseObject("true")), true);
-  EXPECT_EQ(bool(FirebaseObject("false")), false);
-  EXPECT_EQ(int(FirebaseObject("42")), 42);
-  EXPECT_EQ(float(FirebaseObject("43.0")), 43.0);
-  EXPECT_EQ(String(FirebaseObject("\"foo\"")), "foo");
+TEST(FirebaseObjectTest, GetBool) {
+  FirebaseObject obj("true");
+  EXPECT_EQ(true, obj.getBool());
+  EXPECT_TRUE(obj.success());
+  EXPECT_FALSE(obj.failed());
+  EXPECT_EQ(obj.error(), "");
 }
 
-TEST(FirebaseObjectTest, JsonObject) {
+TEST(FirebaseObjectTest, GetInt) {
+  FirebaseObject obj("42");
+  EXPECT_EQ(42, obj.getInt());
+  EXPECT_TRUE(obj.success());
+  EXPECT_FALSE(obj.failed());
+  EXPECT_EQ(obj.error(), "");
+}
+
+TEST(FirebaseObjectTest, GetFloat) {
+  FirebaseObject obj("43.0");
+  EXPECT_EQ(43, obj.getFloat());
+  EXPECT_TRUE(obj.success());
+  EXPECT_FALSE(obj.failed());
+  EXPECT_EQ(obj.error(), "");
+}
+
+TEST(FirebaseObjectTest, GetString) {
+  FirebaseObject obj("\"foo\"");
+  EXPECT_EQ("foo", obj.getString());
```

Comment:
GetObject tests that /foo is equal to "bar" But that isn't actually testing that we can parse more than one level of a path. The path parsing, as I mention above is non-obvious code so it should probably have some tests that ensure it is working properly. 
If we tested a path of "/foo/bar" gave a value of "baz" that would probably be enough. Although I would still want to know how it handled edge cases like if I gave a path of "/foo/bar/" (with a trailing /).